File: forumClient/src/Dto/app/CategoryInfo.ts
import type { BoardBasicInfo } from "./BoardInfo"
import type { TopicBasicInfo } from "./TopicInfo"

export type CategoryEntry = {
    categoryId: string,
    categoryName: string,
    categoryDescription: string
}

export type CategoryBasicInfo = {
    category: CategoryEntry,
    boards: BoardBasicInfo[],
}

export type CategoryFullInfo = {
    category: CategoryEntry,
    boards: BoardBasicInfo[],
    totalBoards: number
}

export type CreateCategoryRequest = {
    categoryName: string,
    categoryDescription: string
}

export type CategorySummary = {
    totalCategories: number
}

File: server/RestApiServer/Dto/App/CategoryInfo.cs
using RestApiServer.Db;
using RestApiServer.Db.Users;
using RestApiServer.Utils;

namespace RestApiServer.Dto.App
{
    public class CategoryBasicInfo
    {
        public required CategoryEntry Category { get; set; } = null!;
        public required List<BoardBasicInfo> Boards { get; set; } = new();
    }

    public class CategoryFullInfo
    {
        public required CategoryEntry Category { get; set; } = null!;
        public required List<BoardBasicInfo> Boards { get; set; } = new();
        public required int TotalBoards { get; set; } = 0;
    }
}